CardanoSharp.Wallet.Test/TransactionTests.cs

```diff
@@ -155,25 +305,173 @@ public void DeserializeMultiAssetTransaction()
+                    .SetFee(171397)
+                    .SetTtl(57910820)
+                    .SetCertificate(CertificateBuilder.Create
+                        .SetStakeRegistration(stakeHash)
+                        .SetStakeDeregistration(stakeHash)
+                        .SetStakeDelegation(stakeHash, stakeHash))
+                )
+                .SetWitnesses(witnesses)
+                .SetAuxData(auxData)
+                .Build();
+
+            // Assert the TransactionBody values are expected
+            // Assert the TransactionBodyInputs values are expected
+            List<TransactionInput> actualTransactionInputs = new List<TransactionInput> {
+                new TransactionInput { TransactionId = input1TxHash, TransactionIndex = 1 }
+            };
+            Assert.Equal(expected.TransactionBody.TransactionInputs.Count, actualTransactionInputs.Count);
+
+            // There is only 1 transaction input in the expected transaction body
+            TransactionInput expectedTransactionInput = expected.TransactionBody.TransactionInputs.First();
+            TransactionInput actualTransactionInput = actualTransactionInputs.First();
+            Assert.Equal(expectedTransactionInput.TransactionId, actualTransactionInput.TransactionId);
+            Assert.Equal(expectedTransactionInput.TransactionIndex, actualTransactionInput.TransactionIndex);
+
+            // Assert the TransactionBodyOutputs values are expected
+            Dictionary<byte[], ulong> nativeAssetToken = new Dictionary<byte[], ulong> { { "00010203".HexToByteArray(), 60 }, { "00010204".HexToByteArray(), 240 } };
+            List<TransactionOutput> actualTransactionOutputs = new List<TransactionOutput> {
+                new TransactionOutput { Address = payment1Addr.GetBytes(),
+                                        Value = new TransactionOutputValue { Coin = 1, MultiAsset = new Dictionary<byte[], NativeAsset> { { getGenesisPolicyId(), new NativeAsset { Token = nativeAssetToken } } } }
+                }
+            };
+            Assert.Equal(expected.TransactionBody.TransactionOutputs.Count, actualTransactionOutputs.Count);
+
+            // There is only 1 transaction output in the expected transaction body
+            TransactionOutput expectedTransactionOutput = expected.TransactionBody.TransactionOutputs.First();
+            TransactionOutput actualTransactionOutput = actualTransactionOutputs.First();
+            Assert.Equal(expectedTransactionOutput.Address, actualTransactionOutput.Address);
+            Assert.Equal(expectedTransactionOutput.Value.Coin, actualTransactionOutput.Value.Coin);
+            Assert.Equal(expectedTransactionOutput.Value.MultiAsset.Count, actualTransactionOutput.Value.MultiAsset.Count);
+            Assert.Equal(expectedTransactionOutput.Value.MultiAsset.Keys.ToList(), new List<byte[]> { getGenesisPolicyId() });
+
+            // Dictionary keys byte[] are compared by reference not by value
```

Comment:
I commented here that I needed to get the reference values of the byte arrays rather than the actual values in order for the dictionary to say that the key was in the dictionary.

I'm not sure that's what we want but I don't think it should be changed in this PR